### PR TITLE
feat[lang]: support `block.blobbasefee`

### PR DESCRIPTION
**Pull request by tserg**
### What I did

Support `block.blobbasefee`.

### Commit message

```
feat: support `block.blobbasefee`

This PR adds support for `block.blobbasefee`, which was introduced
in EIP-4844.
```
### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/venom/ir_node_to_venom.py [hunk 1]
```diff
@@ -75,7 +76,6 @@
         "extcodehash",
         "balance",
         "msize",
-        "basefee",
```
Comment (charles-cooper) [inline vyper/venom/ir_node_to_venom.py:78]: why is this removed?

File: vyper/evm/opcodes.py [hunk 2]
```diff
@@ -75,6 +75,8 @@
     "CHAINID": (0x46, 0, 1, 2),
     "SELFBALANCE": (0x47, 0, 1, 5),
     "BASEFEE": (0x48, 0, 1, 2),
+    "BLOBHASH": (0x49, 1, 1, 3),
```
Comment (charles-cooper) [inline vyper/evm/opcodes.py:78]: should we also support this? i guess it should probably be supported similarly to blockhash

File: vyper/venom/ir_node_to_venom.py (hunk 1 above)
Comment (tserg) [reply vyper/venom/ir_node_to_venom.py:78]: I think this is a duplicate since there is already a "basefee" at L50.

File: vyper/evm/opcodes.py (hunk 2 above)
Comment (tserg) [reply vyper/evm/opcodes.py:78]: Yes, I think we should. I have done some work on it locally but I am still figuring out how to test it. How about moving this to a separate PR so it does not block `block.blobbasefee`?

File: tests/functional/syntax/test_block.py [hunk 3]
```diff
@@ -153,3 +153,31 @@ def foo() -> uint256:
...
+    a: uint256 = block.blobbasefee
+    return a
+    """,
+]
+
+
+@pytest.mark.requires_evm_version("cancun")
+@pytest.mark.parametrize("good_code", valid_list)
+def test_block_blob_success(good_code):
+    assert compiler.compile_code(good_code) is not None
```
Comment (charles-cooper) [inline tests/functional/syntax/test_block.py:183]: do we need to actually test the behavior? or at least check that blobbasefee is in the assembly

File: vyper/evm/opcodes.py [hunk 4]
```diff
@@ -75,6 +75,7 @@
     "CHAINID": (0x46, 0, 1, 2),
     "SELFBALANCE": (0x47, 0, 1, 5),
     "BASEFEE": (0x48, 0, 1, 2),
+    "BLOBBASEFEE": (0x4A, 0, 1, 2),
```
Comment (pcaversaccio) [inline vyper/evm/opcodes.py:78]:
```suggestion
    "BLOBBASEFEE": (0x4A, 0, 1, (None, None, None, 2)),
```

File: tests/functional/codegen/environment_variables/test_blobbasefee.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    assert c.get_blobbasefee() == MIN_BLOB_BASE_FEE
+
+    a0 = w3.eth.account.from_key(f"0x{'00' * 31}01")
+
+    text = "Vyper is the language of the sneks"
+    encoded_text = abi.encode("(string)", (text,))
+    # Blobs contain 4096 32-byte field elements. Subtract the length of the encoded text
+    # divided into 32-byte chunks from 4096 and pad the rest with zeros.
+    blob_data = (b"\x00" * 32 * (4096 - len(encoded_text) // 32)) + encoded_text
```
Comment (charles-cooper) [inline tests/functional/codegen/environment_variables/test_blobbasefee.py:25]: i think we can use `rjust` or `ljust` for this

File: tests/functional/codegen/environment_variables/test_blobbasefee.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    for _i in range(10):
+        tx = {
+            "type": 3,
+            "chainId": 1337,
+            "from": a0.address,
+            "to": "0xb45BEc6eeCA2a09f4689Dd308F550Ad7855051B5",
```
Comment (charles-cooper) [inline tests/functional/codegen/environment_variables/test_blobbasefee.py:32]: what is this address? (if it's just a random address then we should leave a comment)

File: tests/functional/codegen/environment_variables/test_blobbasefee.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    return block.blobbasefee
+"""
+    c = get_contract_with_gas_estimation(code)
+
+    assert c.get_blobbasefee() == MIN_BLOB_BASE_FEE
+
+    a0 = w3.eth.account.from_key(f"0x{'00' * 31}01")
+
+    text = "Vyper is the language of the sneks"
+    encoded_text = abi.encode("(string)", (text,))
```
Comment (charles-cooper) [inline tests/functional/codegen/environment_variables/test_blobbasefee.py:22]: why does this need to be abi encoded btw?

**Review by charles-cooper — APPROVED** [review]
